cmd/kubeadm/app/cmd/init.go

```diff
@@ -527,3 +529,38 @@ func (d *initData) PatchesDir() string {
 	}
 	return ""
 }
+
+// manageSkippedAddons syncs proxy and DNS "Disabled" status and skipPhases.
+func manageSkippedAddons(cfg *kubeadmapi.ClusterConfiguration, skipPhases []string) []string {
+
+	// If the DNS or Proxy addons are disabled, skip the corresponding phase
+	if cfg.DNS.Disabled && !isPhaseInSkipPhases(kubeadmconstants.AddonPhase, skipPhases) && !isPhaseInSkipPhases(kubeadmconstants.CoreDNSPhase, skipPhases) {
+		skipPhases = append(skipPhases, kubeadmconstants.CoreDNSPhase)
+	}
+	if cfg.Proxy.Disabled && !isPhaseInSkipPhases(kubeadmconstants.AddonPhase, skipPhases) && !isPhaseInSkipPhases(kubeadmconstants.KubeProxyPhase, skipPhases) {
+		skipPhases = append(skipPhases, kubeadmconstants.KubeProxyPhase)
+	}
```

Comment:
```suggestion
	// If the DNS or Proxy addons are disabled, skip the corresponding phase.
	// Alternatively, update the proxy and DNS "Disabled" status based on skipped addon phases.
	var (
		skipDNSPhase = false
		skipProxyPhase = false
	)
	if isPhaseInSkipPhases(kubeadmconstants.AddonPhase, skipPhases) {
		skipDNSPhase = true
		skipProxyPhase = true
		cfg.DNS.Disabled = true
		cfg.Proxy.Disabled = true
	}
	if isPhaseInSkipPhases(kubeadmconstants.CoreDNSPhase, skipPhases) {
		skipDNSPhase = true
		cfg.DNS.Disabled = true
	}
	if isPhaseInSkipPhases(kubeadmconstants.KubeProxyPhase, skipPhases) {
		skipProxyPhase = true
		cfg.Proxy.Disabled = true
	}
	if cfg.DNS.Disabled && !skipDNSPhase {
			skipPhases = append(skipPhases, kubeadmconstants.CoreDNSPhase)
	}
	if cfg.DNS.Disabled && !skipProxyPhase {
			skipPhases = append(skipPhases, kubeadmconstants.KubeProxyPhase)
	}
```

can the whole function be changed to this?

Reply:
Yes, of course. You saved us from checking it again and again.